### PR TITLE
Amp pages

### PR DESCRIPTION
**Pull request by anthotsang**
https://app.asana.com/0/813012001853927/1159492027039817

### CODE REVIEW

File: app/controllers/coin_articles_controller.rb [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
   rescue_from ActiveRecord::RecordNotFound, :with => :record_not_found
   before_action :set_coin_article, only: [:show]
+  prepend_before_action :set_amp, only: [:index]
 
   breadcrumb 'How to Buy Cryptocurrency', :coin_articles_path, match: :exact
 
   def index
     distribute_reads(max_lag: MAX_ACCEPTABLE_REPLICATION_LAG, lag_failover: true) do
       @coin_articles = CoinArticle.page(params[:page]).per(params[:limit])
+      @coin_articles.each # force load
```
Comment (hanchang) [inline app/controllers/coin_articles_controller.rb]: Thanks for the comment! Just curious why force loading is required here?

File: app/views/components/_rollbar.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<script>
```
Comment (hanchang) [inline app/views/components/_rollbar.html.erb]: I'm guessing this is for Rollbar client side JS?

File: config/initializers/rollbar.rb [hunk 3]
```diff
@@ -80,16 +80,5 @@
...
-    accessToken: ENV.fetch('ROLLBAR_CLIENT_ACCESS_TOKEN'),
-    captureUncaught: true,
-    captureUnhandledRejections: true,
-    logLevel: 'error',
-    hostWhiteList: ['coinfi.com', 'herokuapp.com'],
-    payload: {
-      environment: "production"
-    }
-  }
+  config.js_enabled = false
```
Comment (hanchang) [inline config/initializers/rollbar.rb]: Ah interesting, so the Rollbar `js_enabled` functionality doesn't work out of the box and requires custom integration?

File: app/controllers/coin_articles_controller.rb (hunk 1 above)
Comment (anthotsang) [reply app/controllers/coin_articles_controller.rb]: distribute_reads needs to force load to work properly. normally the recommended method is `to_a`, but that would get rid of the methods provided by Kaminari
distribute_reads also suggests `load`, but that didn't seem to be working either

File: app/views/components/_rollbar.html.erb (hunk 2 above)
Comment (anthotsang) [reply app/views/components/_rollbar.html.erb]: Yep

File: config/initializers/rollbar.rb (hunk 3 above)
Comment (anthotsang) [reply config/initializers/rollbar.rb]: No, it works, but the problem was it was enabled for everything and couldn't be selectively disabled.
So, it was being injected into AMP pages, which can't support external javascript.

**Comment by anthotsang** [comment]
> Also, I'm guessing we're only applying AMP to the coin_articles index page at the moment because we're rolling it out slowly to test before doing some sort of sitewide (or at least coin_articles-wide) release?

Yep! Testing bit-by-bit to hopefully get a better idea of what the effects are.

**Pull request merged**